### ISSUE TITLE
Inference on SNN model for event dataset (NMNIST) with weights trained from ANN model on MNIST 

### ISSUE THREAD

**Issue by vankhoa21991** [CEA-LIST/N2D2#36]
Hello, I'm trying an experiment like training an ANN (1 layer FC) with MNIST, and then import the weights to a SNN with same structure to do inference on NMNIST. But there are two many options in the ini file that I don't understand, like should I choose the base model of the SNN as Transcode, or as Spike_Analog. And then there are two channels in the SNN, so the model is not similar.

 I post the two ini file here, could you help me to check it.

the training ini to train ANN with back propagation on MNIST:

; Command:
; ./n2d2.sh "$N2D2_MODELS/LeNet.ini" -learn 6000000 -log 100000

DefaultModel=Frame_CUDA

; Database
[database]
Type=MNIST_IDX_Database
Validation=0.2 ; Use 20% of the dataset for validation

; Environment
[sp]
SizeX=32
SizeY=32
BatchSize=128

[sp.Transformation_1]
Type=ChannelExtractionTransformation
CSChannel=Gray

[sp.Transformation_2]
Type=RescaleTransformation
Width=[sp]SizeX
Height=[sp]SizeY

; Output layer (fully connected)
[fc1]
Input=sp
Type=Fc
NbOutputs=500
ActivationFunction=Rectifier
WeightsFiller=HeFiller
ConfigSection=common.config

[fc2]
Input=fc1
Type=Fc
NbOutputs=10
ActivationFunction=Linear
ConfigSection=common.config

[softmax]
Input=fc2
Type=Softmax
NbOutputs=[fc2]NbOutputs
WithLoss=1

[softmax.Target]

; Common config for static model
[common.config]
NoBias=0
WeightsSolver.LearningRate=0.05
WeightsSolver.Momentum=0.9
WeightsSolver.Decay=0.0005
Solvers.LearningRatePolicy=StepDecay
Solvers.LearningRateStepSize=[sp]_EpochSize
Solvers.LearningRateDecay=0.993
Solvers.Clamping=0.0:1.0

The test ini to import the weight from ANN to use on SNN:

DefaultModel=Transcode_CUDA

; Database
[database]
Type=N_MNIST_Database

; Environment
[env]
SizeX=32
SizeY=32
BatchSize=1
NbChannels=2
ConfigSection=env.config

[env.config]
ReadAerData=1

[env.Transformation_1]
Type=RescaleTransformation
Width=[env]SizeX
Height=[env]SizeY

; Output layer (fully connected)
[fc1]
Input=env
Type=Fc
NbOutputs=500
ConfigSection=common.config
ConfigSection(Spike_Analog)=common_Spike_Analog.config
ConfigSection(Spike_RRAM)=common_Spike_RRAM.config

[fc2]
Input=fc1
Type=Fc
NbOutputs=10
ConfigSection=common.config,fc2.config
ConfigSection(Spike_Analog)=common_Spike_Analog.config
ConfigSection(Spike_RRAM)=common_Spike_RRAM.config

[fc2.Target]

[fc2.config]
; Spike-based computing
TerminateDelta=4
BipolarThreshold=1

; Common config for static model
[common.config]
NoBias=0
WeightsSolver.LearningRate=0.005
Threshold=1.0
Refractory=10,000,000

; Common config for Spike_Analog model
[common_Spike_Analog.config]
EnableStdp=0
Threshold=50.0
StdpLtp=50,000,000
InhibitRefractory=1,000
Leak=100,000,000
BipolarIntegration=0
RefractoryIntegration=0
WeightsMinMean="1.0e-4; 10%"
WeightsMaxMean="1.0; 10%"
WeightsRelInit="0.67; 20%"
WeightIncrement="0.0017; 1%"
WeightIncrementDamping=-3
WeightDecrement="0.0007; 1%"
WeightDecrementDamping=0

; Common config for Spike_RRAM model
[common_Spike_RRAM.config]
EnableStdp=0
Threshold=1.52e-2
LtdProba=0.0007
LtpProba=0.0020
StdpLtp=50,000,000
InhibitRefractory=1,000
Leak=100,000,000
BipolarWeights=0
BipolarIntegration=0
RefractoryIntegration=0
WeightsMinMean="1.19e-5; 1.67e-5"
WeightsMaxMean="3.04e-4; 3.86e-5"
WeightsRelInit="2.0e-4; 20%"
WeightsMaxVarSlope=-1.7575
WeightsMaxVarOrigin=-17.1691
WeightsMinVarSlope=-0.0791
WeightsMinVarOrigin=-1.7081
WeightsSetProba="0.843; 0.197"
WeightsResetProba="0.992; 0.00748"
SynapticRedundancy=64

Thanks

**Comment by EdgarLemaire** [comment]
I'm not quite sure about it, but I guess "Spike_Analog" is made for "physical synapses" (RRAM) simulation. Spike_Analog seems to enable the use of STDP. The many parameters are related to the physical properties of the simulated synapse.
On the other hand, "Transcode" is made to simulate spiking networks at a higher level of abstraction : you do not considerate the hardware synapse, but only the algorithm. With transcode, you can't emulate STDP.

Hope this helps before Olivier Bichler gives a better answer.

Regards

Edgar
